doc/tutorials/nvmx_virtualization.rst

```diff
@@ -0,0 +1,169 @@
+.. _nested_virt:
+
+Enable Nested Virtualization
+############################
+
+With nested virtualization enabled, it's possible to run virtual machine
+instances inside of level 1 guests. Though it seems both nested guests and
+User VMs are launched from the guest VM, the distinction between them is
+that, in the case of User VM, hypervisor is not running on Service OS, and
+besides the device model, ACRN provides all the support to run the User VMs.
+While for nested guests, an instance of guest hypervisor is running on the
+guest to manage the nested VMs.
+
+The conventional single-level virtualization has two levels - the host
+(ACRN hypervisor) and the guest VMs. With nested virtualization enabled,
+ACRN is able to run guests with their associated virtual machines, which
+has three levels:
+
+* The host (ACRN hypervisor), which we call L0 hypervisor, or L0 VMM
+* The guest hypervisor, which we call L1 hypervisor, or L1 VMM
+* The nested guest, which we call L2 guests
+
+.. figure:: images/nested_1.png
+   :width: 700px
+   :align: center
+
+   Generic Nested Virtualization
+
+
+High Level ACRN Nested Virtualization Design
+********************************************
+
+The image below shows the high-level design of nested virtualization in
+ACRN. Nested VMX is enabled by so by allowing a guest to use VMX
+instructions, and emulating them using the single level of VMX available
+in the hardware.
+
+A logical processor uses virtual-machine control data structures (VMCSs) while
+it is in VMX operation. The core of the NVMX emulation is ACRN builds VMCS02
+out of the VMCS12 which is built by L1 hypervisor, to actually run the L2 guest
+
+.. figure:: images/nested_arch_1.png
+   :width: 700px
+   :align: center
+
+   Nested Virtualization in ACRN
+
+
+Restrictions and constrains
+***************************
+
+Currently nested virtualization is considered as an experimental feature.
+As of today, the tested platforms are: Tiger Lake RVP and Kaby Lake NUC.
+
+L1 VMs have the following restrictions:
+
+* KVM is the only L1 hypervisor that supported by ACRN currently
+* KVM runs in 64-bits mode
+* KVM enables EPT for L2 guests
+* QEMU is used to launch L2 guests
+
+Constrains on L1 guest configuration:
+
+* Local APIC passthrough should be enabled
+* use SCHED_NOOP scheduler because ACRN can't receive timer interrupts
+  on LAPIC passthrough pCPUs
+* Don't support hybrid mode currently
+
+L1 VM configuration
+*******************
+
+The nested virtualization feature is disabled by default in ACRN. You can
+enable it using the :ref:`ACRN configuration toolset <acrn_config_workflow>`
+with the following steps:
+
+- Configure system level features:
+
+   - Edit ``NVMX_ENABLED`` to ``y`` in ACRN scenario XML configuration to
+     enable nested virtualization
+
```

Comment:
So which scenario do you use? Industry?

Reply:
Personally I tested with SDC, which is supposed to same with industry, from the perspective of how to support nested virtualization, and logical partition mode, and hybrid.

Reply:
I will add :ref:`acrn_configuration_tool` and mentioned that it's preferred to use GUI config.

Reply:
Added screen shot from the ACRN config GUI.